lib/types.nix

```diff
@@ -236,6 +306,8 @@ rec {
             args = { name = ""; }; }).options;
         getSubModules = opts';
         substSubModules = m: submodule m;
+        # disabling type merging for submodules
+        typeMerge = f: f': null;
```

Comment:
I highly doubt this would work with the current NixOS modules, as we at least use that for the `fileSystems`.

Currently the merging process of submodules is handled by `mergeOptionDecls`, but we still expects to be able to merge the type.  The `opts` value is similar to the list of enumerated values of the `enum` type, but as the submodules are extracted, I guess you could just define the following `functor`:

```nix
{
  functor = (defaultFunctor name) // {
    # Merging of submodules is done as part of mergeOptionDecls, as we have to annotate
    # each submodule with its location.
    payload = [];
    binOp = lhs: rhs: [];
  };
}
```